src/components/ha-icon-picker.ts

```diff
@@ -58,7 +100,9 @@ export class HaIconPicker extends LitElement {
         item-label-path="icon"
         .value=${this._value}
         allow-custom-value
-        .filteredItems=${iconItems}
+        .filteredItems=${guard([this._filterString, ICONS.length], () =>
+          this._filterIcons(this._filterString, ICONS)
+        )}
```

Comment:
Are you sure the guard and the memoize function is needed ? Only one of these should be necessary I think.

Reply:
Yeah they are mostly redundant.  The `guard` has the extra advantage of protecting the template itself.  This will be a moot point when I submit the actual fix. 😉